primitives/src/validator.rs

```diff
@@ -62,7 +62,7 @@ pub struct RejectState {
     pub reason: String,
     pub state_root: String,
     pub signature: String,
-    pub balances: Option<String>,
+    pub balances: Option<BalancesMap>,
```

Comment:
We should think about what is really necessary for RejectState and how to actually add it, as these Options don't tell me anything at this point:
- When should they be present?
- Why is the timestamp an Option?
- In what cases we do have RejectState?

Reply:
RejectState cases -> InvalidSignature, InvalidRootHash etc
The properties of RejectState depend on the State being rejected.

Reply:
If it depends on the State being rejected, why don't we just include the whole message, instead of making variants with `Option` and then in all the code check for those variants?

Reply:
i think we should follow the implementation for now.
Later we can create Adex improvement proposals for that